Fix:

error: implicit declaration of function 'gethostbyaddr_r' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

https://gitlab.torproject.org/tpo/core/torsocks/-/issues/40005#note_2720242

Just declaring the function prototype like this without providing an
implementation should not work because macOS does not have this function,
and yet the test suite passes.
--- tests/test_dns.c.orig	2022-05-25 09:36:05.000000000 -0500
+++ tests/test_dns.c	2024-08-02 20:40:40.000000000 -0500
@@ -26,6 +26,8 @@
 #include <tap/tap.h>
 #include "helpers.h"
 
+int gethostbyaddr_r();
+
 #define NUM_TESTS 6
 
 struct test_host {
